README 파일을 수정하라

eastshine-high opened the pull request:
README는 디스커션에 의견 주신 것을 최대한 반영해서 뼈대를 잡아봤습니다. 프로젝트 진행하면서 내용을 채워보겠습니다.

hannut91 committed:
Define Discount domain model

유비쿼터스 언어를 보고 할인 도메인 모델을 만들었습니다.
할인 도메인 모델은 할인이 할인 대상 상품을 정할 수 있습니다.

hannut91 committed:
Define DiscountRepository

할인 레포지토리를 생성했습니다.

hannut91 committed:
Register discount

할인 관련 기능들을 관리하는 서비스를 만들었습니다.
할인을 등록할 수 있습니다.

hannut91 committed:
Create health check controller

서버가 살아있는지 확인할 수 있도록 Hello!를 반환하는 HelloController를
만들었습니다.

hannut91 committed:
Clear querydsl directory before build

빌드하기 전에 이미 querydsl 폴더가 있다면 삭제하고 빌드하도록 했습니다.

hannut91 committed:
Remove version information

`./gradlew assemble`할 때 버전 정보로 jar파일이 만들어져서, 버전을
변경할 때 마다 Dockerfile을 수정해야 합니다.
그래서 버전정보를 삭제했습니다.

hannut91 committed:
Dockerize project

배포를 위해서 도커 파일을 만들었습니다.

hannut91 committed:
Run GitHub Actions on push and pull request

푸시하고 풀리퀘스트를 만들었을 때 GitHub Actions을 실행하도록 했습니다.

etff committed:
add swagger config

- api 문서화를 위한 도구로 Swagger를 의존성에 추가했습니다.

hannut91 committed:
Change heal check url to /health-check from root

API 루트는 다른 것을 보여줄 수 있는 경우가 있어서, 헬스 체크 url을 따로
사용하도록 수정했습니다.

hannut91 committed:
Rename to apply from attach

할인을 적용할 수 있다로 표현하기로 하여 apply로 변경합니다.

hannut91 committed:
Reveal ubiquitous language in test names

테스트 이름이 너무 기능적인 부분만 확인하는 것 같아서, 유비쿼터스 언어가
드러날 수 있도록 변경했습니다.

etff committed:
git keep 제거

eastshine-high committed:
README 파일을 수정하라

hannut91 commented:
풀 리퀘스트 제목이 아쉽습니다. 풀 리퀘스트의 제목에는 작업의 내용이 아닌, 목적을 작성하는게 좋습니다. 

`수정`이라는 말은 너무 많은 의미로 해석될 수 있습니다. 변경을 한것도 수정이고, 삭제도 수정이고, 추가한 것도 수정이라고 해석될 수 있고 여러가지 의미로 해석될 수 있습니다.

아마도 이 작업에 여러 작업이 함께 있어서 그런 것 같아요. 만약 README에 유비쿼터스 언어만 추가 했다면

`README에 유비쿼터스 언어를 추가하라`라고 되었겠죠

etff reviewed (changes requested):
고생하셨습니다. 몇가지 생각해볼만한 코멘트를 남겼습니다.

문서를 추가로 작업해주는 것도 좋지만 연습이라고 생각하고
처음에 잘 정리해두면 인수인계를 받는 사람이 많은 도움을 받을 수 있을 것같습니다.

eastshine-high commented:
> 풀 리퀘스트 제목이 아쉽습니다. 풀 리퀘스트의 제목에는 작업의 내용이 아닌, 목적을 작성하는게 좋습니다.
> 
> `수정`이라는 말은 너무 많은 의미로 해석될 수 있습니다. 변경을 한것도 수정이고, 삭제도 수정이고, 추가한 것도 수정이라고 해석될 수 있고 여러가지 의미로 해석될 수 있습니다.
> 
> 아마도 이 작업에 여러 작업이 함께 있어서 그런 것 같아요. 만약 README에 유비쿼터스 언어만 추가 했다면
> 
> `README에 유비쿼터스 언어를 추가하라`라고 되었겠죠

네 풀 리퀘스트의 제목은 목적을 작성하는 게 좋군요. 수정이란 표현은 조심해서 사용해야 겠네요. 감사합니다!

JamieShin0201 committed:
Handle discount registration request

할인 등록 정보와 함께 POST /discounts 요청 시 할인을 등록합니다.

etff committed:
정책을 추가하라

- 정책에 정률할인과 정액할인을 추가한다

etff committed:
기간을 추가하라

etff committed:
쿠폰번호를 추가하라

정률을 비율을 추가하라

쿠폰 번호 테스트를 추가한다

비율의 동등성 수정과 인자를 수정하라

쿠폰의 수량을 추가하라

매직넘버를 제거한다

기간 빈 값 입력시 예외 처리

할인 값을 추가한다

etff committed:
쿠폰 도메인을 추가한다

쿠폰 이름을 추가한다

쿠폰 영속화기능을 추가한다

쿠폰 정책 기능을 추가하라

- 정률 정책을 추가한다.
- 정액 정책을 추가한다.

etff committed:
쿠폰 정책을 상속으로 변경하라

- 정률, 정액 정책으로 변경한다.

쿠폰 구분 컬럼을 수정하라

etff committed:
쿠폰 번호를 쿠폰 코드로 용어를 수정한다.

> https://github.com/CodeSoom/DDD-Kurly-Clone-Promotion/discussions/7#discussioncomment-1513859

etff committed:
정책 생성 테스트를 추가한다

etff committed:
동일성 비교 기능을 추가한다

etff committed:
쿠폰번호를 랜덤생성하지 않고 키워드로서 입력받도록 수정한다

etff committed:
유비쿼터스 언어를 반영하도록 용어를 수정한다

etff committed:
정책 타입의 중복 컬럼을 제거하라

etff committed:
오타수정

etff committed:
유효성 검사를 추가하라

etff committed:
불필요한 여백을 제거합니다

etff committed:
기준 유효성 중복검사를 제거하라

etff committed:
오타를 수정하라

etff committed:
최소 적용 금액을 추가한다

etff committed:
이름과 키워드의 중복검사를 추가하라

etff committed:
잭슨 변환 오류를 수정하라

- getter 설정을 통해 객체를 변환하려는 오류를 수정합니다

eastshine-high committed:
프로젝트 구조를 추가하라

유비쿼터스 언어를 제거하고 프로젝트 구조를 추가하였습니다.
